src/chainlit/frontend/src/components/dataset/filters/AuthorSelect.tsx

```diff
@@ -2,54 +2,50 @@ import { useEffect, useState } from 'react';
 import { useRecoilState, useRecoilValue } from 'recoil';
 
 import Box from '@mui/material/Box';
-import FormControl from '@mui/material/FormControl';
-import InputLabel from '@mui/material/InputLabel';
-import MenuItem from '@mui/material/MenuItem';
-import Select, { SelectChangeEvent } from '@mui/material/Select';
+import { SelectChangeEvent } from '@mui/material/Select';
+
+import SelectInput, { SelectItem } from 'components/selectInput';
 
 import { clientState } from 'state/client';
 import { datasetFiltersState } from 'state/dataset';
-import { IMember, roleState } from 'state/user';
+import { roleState } from 'state/user';
 
 export default function AuthorSelect() {
   const client = useRecoilValue(clientState);
   const role = useRecoilValue(roleState);
   const [df, setDf] = useRecoilState(datasetFiltersState);
-  const [members, setMembers] = useState<IMember[]>();
+  const [members, setMembers] = useState<SelectItem[]>();
 
   useEffect(() => {
-    client.getProjectMembers().then((res) => setMembers(res));
+    client.getProjectMembers().then((res) => {
+      const members = res.map((member) => ({
+        value: member.email,
+        label: member.email
+      }));
+
+      setMembers([{ value: 'All', label: 'All' }, ...members]);
+    });
   }, [client]);
 
   if (!members || role === 'USER') {
     return null;
   }
 
   const handleChange = (event: SelectChangeEvent) => {
-    const value = event.target.value as string;
+    const value = event.target.value;
     const authorEmail = value === 'All' ? undefined : value;
     setDf({ ...df, authorEmail });
   };
 
   return (
     <Box sx={{ width: 200 }}>
-      <FormControl fullWidth>
-        <InputLabel id="author-filter-select">Author</InputLabel>
-        <Select
-          labelId="author-filter-select"
-          value={df.authorEmail || 'All'}
-          label="Author"
-          onChange={handleChange}
-          size="small"
-        >
-          <MenuItem value={'All'}>All</MenuItem>
-          {members.map((m) => (
-            <MenuItem key={m.email} value={m.email}>
-              {m.email}
-            </MenuItem>
-          ))}
-        </Select>
-      </FormControl>
+      <SelectInput
```

Comment:
The fixed width here seems to be a problem when the option is lengthy
<img width="395" alt="Screenshot 2023-07-13 at 17 23 34" src="https://github.com/Chainlit/chainlit/assets/13104895/777a97f1-5915-4f32-9458-2ff67f5d7e4f">